packages/frontend/src/utils/fieldValidators.ts

```diff
@@ -36,13 +40,38 @@ export const isValidEmail: FieldValidator<string> = (fieldName) => (value) =>
 
 export const isValidEmailDomain: FieldValidator<string[]> =
     (fieldName) => (value) => {
-        if (value) {
-            const hasInvalidValue = value.some((item: string) =>
-                item.match(/@/),
-            );
-            return hasInvalidValue
-                ? `${fieldName} should not contain @, eg: (gmail.com)`
-                : undefined;
+        if (!Array.isArray(value) || !value?.length) {
+            return undefined;
+        }
+
+        const hasInvalidValue = value.some((item: string) => item.match(/@/));
+
+        if (hasInvalidValue) {
+            return `${fieldName} should not contain @, eg: (gmail.com)`;
+        }
+    };
+
+export const isValidOrganizationDomain: FieldValidator<string[]> =
+    (_) => (value) => {
+        if (!Array.isArray(value) || !value?.length) {
+            return undefined;
+        }
+
+        const invalidDomains = value.filter((domain) =>
+            isEmailProviderDomain(domain),
+        );
+
+        if (!!invalidDomains.length) {
+            const readableDomainList = new Intl.ListFormat('en-US', {
+                style: 'long',
+                type: 'conjunction',
+            }).format(invalidDomains);
+
+            return `
+                ${readableDomainList} ${
+                invalidDomains.length === 1 ? 'is' : 'are'
+            } not allowed as organization email
+            `;
```

Comment:
this chunk exists both in the frontend and the backend.
can you extract both and move it to `packages/commmon/src/utils/email.ts`?

maybe the whole validation not only this chunk.

Reply:
looks great, im gonna run tests locally and merge 🥇 🥇 🥇 
thanks so much!